frontend/src/toolbar/product-tours/productToursLogic.ts

```diff
@@ -713,8 +716,27 @@ export const productToursLogic = kea<productToursLogicType>([
 
     events(({ actions, values, cache }) => ({
         afterMount: () => {
+            // Watch for DOM changes to update highlights when elements appear/disappear
+            let mutationTimeout: ReturnType<typeof setTimeout> | null = null
```

Comment:
shouldn't this mutationTimeout also be on the cache? then. we should also clear it in the `beforeUnmount` as well. makes sense?